modules/system/assets/js/snowboard/main/Snowboard.js

```diff
@@ -437,7 +436,6 @@ export default class Snowboard {
         const listeners = this.listensToEvent(eventName);
         if (listeners.length === 0) {
             this.debug(`No listeners found for global promise event "${eventName}"`);
-            return Promise.resolve();
```

Comment:
Is this intentional?

Reply:
Yes, it short-circuits the event listeners if there are none available.